lib/iris/fileformats/netcdf/saver.py

```diff
@@ -1846,6 +1286,8 @@ def record_dimension(names_list, dim_name, length, matching_coords=[]):
             matches the earlier finding.
 
             """
+            if matching_coords is None:
+                matching_coords = []
```

Comment:
Because in the original code the arg `matching_coords=[]` is a mutable default,
which is just a Bad Idea :tm: 
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments